### ISSUE TITLE
Text between multiple tabs needs to be synced

### ISSUE THREAD

**Issue by teesloane** [teesloane/husk#1]
If a user opens up two new tabs, text between them should be synced. 

# Current Save Situation:

- the text in the new tab is saved on:
  - if `x` seconds has passed on keyUp (without another key being pressed down)
  - when the tab is closed.

# Problem:

- user opens a new tab. Writes some stuff. (TabA)
- Open another new tab (TabB): 
  - The stuff last typed in TabA need to be loaded into TabB, so that any writing in TabB _does not_ overwrite local storage:
  - on close tab, a save is executed --> whichever tab is closed last overwrites local storage.

# notes / ideas:

## Ideal solution (haven't researched)

- find out if a function can be executed when a user switches tabs (as if the `window` object loses "focus")
  - if so, save to LS on tab switch... 
  - load LS note contents on tab focus. 
  - basically: a user opens a new tab --> save the old version of Husk --> on tab focus / a new instance of Husk - load information from LS. 
- looks like the [page visibility api](https://developer.mozilla.org/en-US/docs/Web/API/Page_Visibility_API?redirectlocale=en-US&redirectslug=DOM%2FUsing_the_Page_Visibility_API) might do the trick.

 ## other options:

- change the vue data object for the editor to include a `date` property.
  - every time a save event happens, update the date  with `Date.now()`.
  - on loading the tab for the first time, create a time stamp. Diff the document time stamp against the time stamp in local storage?
  - intermittently do a diff on the local storage object against the editors
